### ISSUE TITLE
[Docs]: Several suggestions around docs on dedicated web workers 

### ISSUE THREAD

**Issue by karyon** [microsoft/playwright#29900]
### Page(s)

https://playwright.dev/docs/api/class-worker
https://playwright.dev/docs/api/class-page#page-workers

### Description

My use case: I want to test a feature that only runs in a worker since it uses the [synchronous OPFS API](https://developer.mozilla.org/en-US/docs/Web/API/FileSystemFileHandle/createSyncAccessHandle). So I want to start a worker in a test and evaluate code in there. Since this is more of a unit test, I understand this might be out of scope. I'd like to suggest multiple improvements to the docs:

1. The example on https://playwright.dev/docs/api/class-worker says how to use `page.on('worker', ...)`, but actually making it useful (in particular waiting for it to be created) would be more involved, something like this: 
```js
let worker;
let workerPromise = new Promise((resolve) => {
    page.on('worker', (w) => {
        worker = w;
        resolve();
    });
});

await page.evaluate((workerScriptUrl) => {
    new Worker(workerScriptUrl);
}, SCRIPT_URL);

await workerPromise;
```

2. It uses `page.on('worker', ...)` and `page.workers()` without linking to them, having links would be nice.

3. Having found the Worker class in playwright, I was looking for ways to create one directly e.g. with a constructor or factory method. I understand now that this is not possible, but the docs could have made that clearer by something like "This class is used to represent web workers as returned by page.workers() or the onWorker event" (with links ;) ) .

4. In the end I'm using this:

```js
await page.evaluate((workerScriptUrl) => {
    new Worker(workerScriptUrl);
}, SCRIPT_URL);
let worker;
await poll(() => worker = page.workers()[0]);
```
which seemed to be the shortest way to create and use a worker. The `poll` is necessary since `page.workers()` contains the worker only after a short while. That was surprising to me at first, it would've been nice if the docs warned me of that.

**Comment by mxschmitt** [comment]
Playwright APIs follow a common Node.js design pattern, in this scenario its using event emitters. We expect the user that he's familiar with concepts like Promises and waitFor in order to wait for, in this case a worker, to be created. You could also do something like this which would not require poll and the racy `page.workers()` at all:

```ts
const workerPromise = page.waitForEvent("worker");
await page.evaluate((workerScriptUrl) => {
    new Worker(workerScriptUrl);
}, SCRIPT_URL);
const worker = await workerPromise;
console.log(worker.url())
```

Since these use-cases are pretty specific, I think there is no action item from our side here, if you have concrete suggestions to our docs, feel free to create a Pull Request.
